signing/src/main/java/tech/pegasys/web3signer/signing/config/metadata/parser/YamlSignerParser.java

```diff
@@ -48,12 +51,16 @@ public YamlSignerParser(final Collection<AbstractArtifactSignerFactory> signerFa
   @Override
   public List<ArtifactSigner> parse(final String fileContent) {
     try {
-      final SigningMetadata metaDataInfo =
-          YAML_MAPPER.readValue(fileContent, SigningMetadata.class);
+      final List<SigningMetadata> metadataInfoList = readSigningMetadata(fileContent);
 
-      return signerFactories.stream()
-          .filter(factory -> factory.getKeyType() == metaDataInfo.getKeyType())
-          .map(metaDataInfo::createSigner)
+      return metadataInfoList.stream()
+          .flatMap(
+              metadataInfo ->
+                  signerFactories.stream()
+                      .filter(
+                          factory ->
+                              Objects.equals(factory.getKeyType(), metadataInfo.getKeyType()))
```

Comment:
What's the benefit of using Object.equals for the enum?

Reply:
no benefit, using `==` or `Object.equals` would result in same result.